crates/conformance-test-utils/src/hive/genesis.rs

```diff
@@ -23,6 +36,125 @@ impl HiveGenesisConfig {
+        loader.contract_classes.push((
+            HexFelt(*class_hash),
+            ContractClass::Path {
+                // Add the compiled path to the Kakarot contract filename
+                path: compiled_path.join(filename).with_extension("json").into_os_string().into_string().unwrap(), /* safe unwrap,
+                                                                                             * valid path */
+                version: 0,
+            },
+        ));
+        // Add Kakarot contracts (address, class_hash) to Kakarot Contracts HashMap
+        // Remove .json from filename to get contract name
+        // 36865 is over 0x9000
+        kakarot_contracts.insert(filename.replace(".json", ""), (FieldElement::from(36865 + index), *class_hash));
+    });
+
+    // Get Kakarot contracts address and proxy class hash
+    let kakarot_address = kakarot_contracts.get("kakarot").unwrap().0;
+    let account_proxy_class_hash = kakarot_contracts.get("proxy").unwrap().1;
+
+    // Add Kakarot contracts to Loader
+    // Convert the HashMap to Vec and sort by key to ensure deterministic order
+    let mut kakarot_contracts: Vec<(String, (FieldElement, FieldElement))> = kakarot_contracts.into_iter().collect();
+    kakarot_contracts.sort_by_key(|(name, (_, _))| name.clone());
+    kakarot_contracts.iter().for_each(|(_, (address, class_hash))| {
+        loader.contracts.push((HexFelt(*address), HexFelt(*class_hash)));
+    });
+
+    // Add Hive accounts to loader
+    // Convert the EVM accounts to Starknet accounts using compute_starknet_address
+    // Sort by key to ensure deterministic order
+    let mut hive_accounts: Vec<(reth_primitives::H160, AccountInfo)> = hive_genesis.alloc.into_iter().collect();
+    hive_accounts.sort_by_key(|(address, _)| *address);
+    hive_accounts.iter().for_each(|(evm_address, account_info)| {
+        // Use the given Kakarot contract address and declared proxy class hash for compute_starknet_address
+        let starknet_address = compute_starknet_address(
+            kakarot_address,
+            account_proxy_class_hash,
+            FieldElement::from_byte_slice_be(evm_address.as_bytes()).unwrap(), /* safe unwrap since evm_address
+                                                                                * is 20 bytes */
+        );
+        // Push to contracts
+        loader.contracts.push((HexFelt(starknet_address), HexFelt(account_proxy_class_hash)));
+
+        // Set the balance of the account
+        // Call genesis_fund_starknet_address util to get the storage tuples
+        let balance_storage_tuples = genesis_fund_starknet_address(starknet_address, account_info.balance);
+        balance_storage_tuples.iter().for_each(|balance_storage_tuple| {
+            loader.storage.push(unsafe {
+                std::mem::transmute::<((Felt, Felt), Felt), ((HexFelt, HexFelt), HexFelt)>(*balance_storage_tuple)
```

Comment:
nice!

Reply:
Thanks for the inspiration for `transmute` magic 🪄